src/Gui/DlgSettingsWorkbenchesImp.cpp

```diff
@@ -0,0 +1,475 @@
+
+    buildWorkbenchList();
+
+    //We set the startup setting after building the list so that we can put only the enabled wb.
+    setStartWorkbenchComboItems();
+}
+
+void DlgSettingsWorkbenchesImp::onLoadClicked(const QString &wbName)
+{
+    // activate selected workbench
+    Workbench* originalActiveWB = WorkbenchManager::instance()->active();
+    Application::Instance->activateWorkbench(wbName.toStdString().c_str());
+    Application::Instance->activateWorkbench(originalActiveWB->name().c_str());
+
+    // replace load button with loaded indicator
+    for (int i = 0; i < ui->wbList->count(); i++) {
+        QWidget* widget = ui->wbList->itemWidget(ui->wbList->item(i));
+        if (widget && widget->objectName() == wbName) {
+            QWidget* loadLabel = widget->findChild<QWidget*>(loadLabelStr);
+            QWidget* loadButton = widget->findChild<QWidget*>(loadButtonStr);
+            loadButton->setVisible(false);
+            loadLabel->setVisible(true);
+            break;
+        }
+    }
+}
+
+void DlgSettingsWorkbenchesImp::onWbActivated(const QString &wbName, bool checked)
+{
+    // activate/deactivate the widgets
+    for (int i = 0; i < ui->wbList->count(); i++) {
+        QWidget* widget = ui->wbList->itemWidget(ui->wbList->item(i));
+        if (widget && widget->objectName() == wbName) {
+            QWidget* iconLabel = widget->findChild<QWidget*>(iconLabelStr);
+            QWidget* nameLabel = widget->findChild<QWidget*>(nameLabelStr);
+            QWidget* loadLabel = widget->findChild<QWidget*>(loadLabelStr);
+            QWidget* loadButton = widget->findChild<QWidget*>(loadButtonStr);
+            QCheckBox* autoloadCheckbox = widget->findChild<QCheckBox*>(autoloadCheckboxStr);
+            if (!iconLabel || !nameLabel || !loadLabel || !loadButton || !autoloadCheckbox)
+                return;
+            iconLabel->setEnabled(checked);
+            nameLabel->setEnabled(checked);
+            loadLabel->setEnabled(checked);
+            loadButton->setEnabled(checked);
+            autoloadCheckbox->setEnabled(checked);
+            if (!checked) //disabling wb disable auto-load.
+                autoloadCheckbox->setChecked(false);
+            break;
+        }
+    }
+
+    // Reset the start combo items.
+    setStartWorkbenchComboItems();
+}
+
+/**
+Build the list of unloaded workbenches.
+*/
+void DlgSettingsWorkbenchesImp::buildWorkbenchList()
+{
+    QStringList workbenches = Application::Instance->workbenches();
+    QStringList enabledWbs = getEnabledWorkbenches();
+    QStringList disabledWbs = getDisabledWorkbenches();
+
+    ui->wbList->setDragDropMode(QAbstractItemView::InternalMove);
+    ui->wbList->setSelectionMode(QAbstractItemView::SingleSelection);
+    ui->wbList->viewport()->setAcceptDrops(true);
+    ui->wbList->setDropIndicatorShown(true);
+    ui->wbList->setDragEnabled(true);
+    ui->wbList->setDefaultDropAction(Qt::MoveAction);
```

Comment:
Why not done in the constructor ?

Reply:
No reasons, it was just where those were before. I'll move them to ctor it's better.